x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -21,6 +22,70 @@ interface BuildEventsSearchQuery {
   trackTotalHits?: boolean;
 }
 
+const buildTimeRangeFilter = ({
+  to,
+  from,
+  timestampOverride,
+}: {
+  to: string;
+  from: string;
+  timestampOverride?: string;
+}): estypes.QueryDslQueryContainer => {
+  // If the timestampOverride is provided, documents must either populate timestampOverride with a timestamp in the range
+  // or must NOT populate the timestampOverride field at all and `@timestamp` must fall in the range.
+  // If timestampOverride is not provided, we simply use `@timestamp`
+  return timestampOverride != null
+    ? {
+        bool: {
+          minimum_should_match: 1,
+          should: [
+            {
+              range: {
+                [timestampOverride]: {
+                  lte: to,
+                  gte: from,
+                  format: 'strict_date_optional_time',
+                },
+              },
+            },
+            {
+              bool: {
+                filter: [
+                  {
```

Comment:
Minor nit, but this `@timestamp` range query is reused in the fallback case below, so we could make this a little more DRY by storing in a const.